src/Models/Media.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace RonasIT\Media\Models;
+
+use App\Models\User;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Support\Facades\Auth;
+use RonasIT\Support\Traits\ModelTrait;
+
+class Media extends Model
+{
+    use ModelTrait;
+    use SoftDeletes;
+    use HasFactory;
+
+    protected $fillable = [
+        'link',
+        'name',
+        'owner_id',
+        'is_public',
+        'meta'
+    ];
+
+    protected $casts = [
+        'is_public' => 'boolean',
+        'deleted_at' => 'date',
+        'meta' => 'array',
+    ];
+
+    protected $hidden = ['pivot'];
+
+    public function scopeApplyMediaPermissionRestrictions(Builder $query): void
```

Comment:
please remove it from the model and repository classes and implement this logic using setting special filters in the service class

```php
if (!Auth::check()) {
    $filters['is_public'] = true;
} else {
    $filters['owner_id'] = Auth::id();
}
```

Reply:
Done.

Reply:
why this method still here?

Reply:
Fixed.